src/software/ai/hl/stp/play/enemy_ball_placement_play.cpp

```diff
@@ -56,10 +58,41 @@ void EnemyBallPlacementPlay::ballPlacementWithShadow(
         tactics_to_run[0].emplace_back(crease_defenders[1]);
         tactics_to_run[0].emplace_back(crease_defenders[2]);
 
+        Vector ball_to_net =
+            (world.ball().position() - world.field().friendlyGoalCenter())
+                .normalize(-0.75 - ROBOT_MAX_RADIUS_METERS);
+
         Vector placement_to_net = (placement_point - world.field().friendlyGoalCenter())
                                       .normalize(-0.75 - ROBOT_MAX_RADIUS_METERS);
+
+        // If the enemy hasn't reached the ball yet, we use this flag to avoid shadowing
+        // so that we don't interfere with the enemy robots going to pick up the ball
+        if (!enemy_at_ball)
```

Comment:
why does enemy_ball_placement_play have shadowers? it it so that we can react quickly after the enemy has set the ball at the location?

Reply:
that was the general idea